Parsely/wp-parsely
# JS Unit Testing Framework & coverage for cookie lib

## Description

* Adds npm scripts for unit testing
* Makes the cookies lib testable by wrapping `document.cookie`
* Covers `lib/cookies` basic functionality

## Motivation and Context

Adding a testing framework for our frontend code is integral to being able to confidently make changes.
We can hook this into a `push` action to run alongside our PHP tests.

## How Has This Been Tested?




### Automated tests
`npm run test:unit`

### Manual tests

* Set an API secret on your test site
* Add the recommended content widget
* Browse to the frontend of your test site
* Manually set a cookie to test in the browser console:
    ```
    document.cookie = `_parsely_visitor=${ JSON.stringify( { id: 'gibberish' } ) }`
    ```
* Patch the widget script to `console.log( { uuid } );` [here](https://github.com/Parsely/wp-parsely/blob/db17ace0b0e0fcd9e74c8e4383cd589d778c4289/src/js/recommended-widget/index.js#L31)
* `npm run start`
* Reload the page
* See `{uuid: "gibberish"}` in the console

## Screenshots (if appropriate):

![Screen Shot 2021-05-06 at 9 42 44 PM](https://user-images.githubusercontent.com/1587282/117385953-0966a400-aeb4-11eb-8236-4f97798ecc38.png)


## Types of changes


New dev tools!
